### DISCUSSION TITLE
how to pass jwt token from rtk query ? if tokens are set into cookies.

### DISCUSSION THREAD

**Discussion by deepakchandrawanshi1063** [reduxjs/redux-toolkit#3160] [Q&A]
while using the rtk query. cookies are not passed in every request. in cookies my tokens are set. so, what i can do to pass cookies into rtk queary.

**Comment by phryneas** [comment]
Do you set `{ credentials: 'include' }`?
Do you see any error messages?

**Reply by deepakchandrawanshi1063** [reply]
yes i tried. It gives me error.
![image](https://user-images.githubusercontent.com/41387081/217799680-96f66ab5-462d-4405-9046-e20dc11efa84.png)

**Reply by phryneas** [reply]
That means that the `fetch` call errored out. There is a good chance that you have a `fetch`-related error message in your browser devtools.

**Reply by deepakchandrawanshi1063** [reply]
no it does not tells me anything.

![image](https://user-images.githubusercontent.com/41387081/217802812-db28bbe5-278d-4663-b0cd-1d12e26239cc.png)

request headers
![image](https://user-images.githubusercontent.com/41387081/217802995-c9bc0316-c7e1-4380-a7a9-cc471f730e1f.png)

payloads
![image](https://user-images.githubusercontent.com/41387081/217803091-4f1e2dec-a744-4616-b7fc-065fe1ecb952.png)

response
![image](https://user-images.githubusercontent.com/41387081/217803173-f04a7ab2-9ade-4d24-a57d-e9553a729204.png)

**Reply by phryneas** [reply]
It very clearly says "CORS error" there. Your server is not configured to accept requests from your client.

**Reply by deepakchandrawanshi1063** [reply]
but it is  already configured in server side.
![image](https://user-images.githubusercontent.com/41387081/217806561-8e182e87-46dd-4668-9834-d04b1999b891.png)

**Reply by phryneas** [reply]
All I can say is that you can clearly see in that screenshot that a CORS preflight request is made, failed and then the request itself failed. You will have to take over the debugging from there. Maybe also just make a request by hand using `fetch` first. RTK Query uses `fetch` internally - if you can make a successful request with `fetch`, you can use the exact same options with RTK Query.

**Reply by deepakchandrawanshi1063** [reply]
thanks. but now solved the issue by using.

cors config
![image](https://user-images.githubusercontent.com/41387081/217807461-56a9b63c-bea9-4258-ab2f-bbf1bb922288.png)

rtk query config
![image](https://user-images.githubusercontent.com/41387081/217807635-e795256a-d3a7-467e-9a45-ab0864295946.png)
